tutormfe/patches/openedx-lms-development-settings

```diff
@@ -17,3 +17,29 @@ CORS_ORIGIN_WHITELIST.append("http://{{ MFE_HOST }}:{{ app["port"] }}")
 LOGIN_REDIRECT_WHITELIST.append("{{ MFE_HOST }}:{{ app["port"] }}")
 CSRF_TRUSTED_ORIGINS.append("{{ MFE_HOST }}:{{ app["port"] }}")
 {% endfor %}
+
+# Start of Dynamic config API settings
+
+ENABLE_MFE_CONFIG_API = True
+
+MFE_CONFIG = {
+    "BASE_URL": "{{ MFE_HOST }}",
+    "CREDENTIALS_BASE_URL": "/csrf/api/v1/token",
+    "DISCOVERY_API_BASE_URL": "{% if DISCOVERY_HOST is defined %}http://{{ DISCOVERY_HOST }}:8381{% endif %}",
+    "ENABLE_NEW_RELIC": False,
```

Comment:
`ENABLE_NEW_RELIC` Probably no need for it to be here, I don't think it's used in runtime but I am not tottaly sure

Reply:
I think it's fine to remove it.  If anybody needs it they can write a plugin. 🤷🏼‍♂️

Reply:
I too would like to keep `ENABLE_NEW_RELIC` out of the plugin, but I can't tell what its default value would be if we don't include it here. The best I can find is [this frontend-build line](https://github.com/openedx/frontend-build/blob/master/config/webpack.prod.config.js#L29) which seems to imply that if `ENABLE_NEW_RELIC` is *anything* except `"false"` (for example, `undefined`), that New Relic will be considered enabled.

Reply:
So then, if we want the false to be effective it shall be defined in `Dockerfile`... I think if we want to let people override what we define as `ENV` then probably we might consider using a combination of `ARG` and `ENV` where what we have as `ENV` now would be the defualt `ARG` and then poeple can change them with `--build-arg`. I am not entrily sure about this path but I think this the pattern @regisb is using with Open edX Dockerfile i.e https://github.com/overhangio/tutor/blob/08a14c80dbf456253f5083f23a1df62f717602a3/tutor/templates/build/openedx/Dockerfile#L18

Reply:
Ah, git blame tells me that our own @regisb was the creator of that setting: https://github.com/openedx/frontend-build/pull/215. Previously, New Relic could not be disabled at all. That is why we (probably) need to explicitly *opt-out* of New Relic integration here, even though you think it'd be an *opt-in* feature.

I think we should leave `"ENABLE_NEW_RELIC": False` in place for now, and then file an upstream frontend issue to make it so New Relic is disabled-by-default.

Reply:
I agree @ghassanmas that it can be defined in the Dockerfile (and removed from the LMS MFE_CONFIG dictionary). I think using an `ARG` would be sufficient, since the variable only needs to be available in the image build environment, not the container's environment.

Reply:
(tangentially, I still think we should file an upstream issue to make NewRelic opt-in instead of opt-out)

Reply:
`ARG` variables are actually available to `RUN` statements in the Dockerfile *and* any of their child processes, I think this would work:
```Dockerfile
ARG DYNAMIC_URL=/something
RUN npm run build  # DYNAMIC_URL will equal "/something" during the build process
```

Reply:
The difference is that `ENV` variables are available in the container when you `docker run` it, whereas `ARG` variables are not.